File: HelpSoftware/Help/Help.Web/Web/src/app/EF/ReviewedArticle.ts
import { User, Article } from "./EF";

export class ReviewedArticle {
  ID_ReviewedArticle: number;
  ReviewedUser: User;
  ArticleReviewed: Article;
  ReviewTime: Date;
  Comment: string;

  deserialize(input): ReviewedArticle {
    this.ID_ReviewedArticle = input.ID_ReviewedArticle;
    this.ReviewedUser = new User().deserialize(input.ReviewedUser);
    this.ArticleReviewed = new Article().deserialize(input.ArticleReviewed);
    this.ReviewTime = input.ReviewTime;
    this.Comment = input.Comment;
    return this;
  }
}

File: HelpSoftware/Help/Help.EF/Entitys/ReviewedArticle.cs
﻿using System;
using System.ComponentModel.DataAnnotations;

namespace Help.EF
{
  public class ReviewedArticle : IHelpEntity
  {
    [Key]
    public virtual int ID_ReviewedArticle { get; set; }

    public virtual User ReviewedUser { get; set; }

    public virtual Article ArticleReviewed { get; set; }

    public virtual DateTime? ReviewTime { get; set; }

    public virtual string Comment { get; set; }

    [Timestamp]
    public virtual byte[ ] TimeStamp { get; set; }

    public string GetEntityName()
    {
      return "ReviewedArticle";
    }

    public int GetID()
    {
      return ID_ReviewedArticle;
    }

    public bool GetWriteHistoryEntry()
    {
      return true;
    }
  }
}